src/librustdoc/clean/mod.rs

```diff
@@ -890,10 +890,10 @@ fn clean_fn_decl_with_args(
 
 fn clean_fn_decl_from_did_and_sig(
     cx: &mut DocContext<'_>,
-    did: DefId,
+    did: Option<DefId>,
     sig: ty::PolyFnSig<'_>,
 ) -> FnDecl {
-    let mut names = if did.is_local() { &[] } else { cx.tcx.fn_arg_names(did) }.iter();
+    let mut names = did.map_or(&[] as &[_], |did| cx.tcx.fn_arg_names(did)).iter();
```

Comment:
Hmm, could this be a behavior change when this is called with an external function? I think if it is, it'd be a bugfix, but I wanted to check. In the inline code path (`inline.rs`), this is called with `Some(did)`, where `did` should be non-local.

Reply:
This should only change code on local functions, external functions always took the else case and are thus equivalent to `Some`. Local function behavior should be affected as an intentional fix related to this PR.

Reply:
You're right, I got confused. Thanks!